OpenApiFilter without Runstage causes NPE

### Describe the bug

In my app I wanted to add a new openapifilter, and did not specify the runstage:
```
@OpenApiFilter
public class MyFilter implements OASFilter {
}
```

I thought this should use the annotations default for the value,
`RunStage value() default OpenApiFilter.RunStage.RUN;`

But instead I just get a NPE on dev mode startup, see below.


Specifying an explicit runstage allows dev mode to start:
```@OpenApiFilter(OpenApiFilter.RunStage.RUN)
public class MyFilter implements OASFilter {
```

### Expected behavior

Default runstage.RUN gets used if nothing else specified

### Actual behavior

$ mvn quarkus:dev
[INFO] Scanning for projects...
[INFO] 
[INFO] ---------------------< org.acme:oas-filter-no-ann >---------------------
[INFO] Building oas-filter-no-ann 1.0.0-SNAPSHOT
[INFO]   from pom.xml
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- quarkus:3.17.8:dev (default-cli) @ oas-filter-no-ann ---
[INFO] Invoking resources:3.3.1:resources (default-resources) @ oas-filter-no-ann
[INFO] Copying 1 resource from src\main\resources to target\classes
[INFO] Invoking quarkus:3.17.8:generate-code (default) @ oas-filter-no-ann
[INFO] Invoking compiler:3.13.0:compile (default-compile) @ oas-filter-no-ann
[INFO] Nothing to compile - all classes are up to date.
[INFO] Invoking resources:3.3.1:testResources (default-testResources) @ oas-filter-no-ann
[INFO] skip non existing resourceDirectory C:\workspaces\mpp\projects\oas-filter-no-ann\src\test\resources
[INFO] Invoking quarkus:3.17.8:generate-code-tests (default) @ oas-filter-no-ann
[INFO] Invoking compiler:3.13.0:testCompile (default-testCompile) @ oas-filter-no-ann
[INFO] Nothing to compile - all classes are up to date.
Listening for transport dt_socket at address: 5005
2025-01-29 10:17:56,538 INFO  [io.qua.dep.dev.IsolatedDevModeMain] (main) Attempting to start live reload endpoint to recover from previous Quarkus startup failure
2025-01-29 10:17:56,985 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (main) Failed to start quarkus: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.smallrye.openapi.deployment.SmallRyeOpenApiProcessor#registerAnnotatedUserDefinedRuntimeFilters threw an exception: java.lang.NullPointerException: Cannot invoke "org.jboss.jandex.AnnotationValue.asEnum()" because the return value of "org.jboss.jandex.AnnotationInstance.value()" is null
        at io.quarkus.smallrye.openapi.deployment.SmallRyeOpenApiProcessor.lambda$getUserDefinedFilters$4(SmallRyeOpenApiProcessor.java:446)
        at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:196)
        at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1709)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:570)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:560)
        at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:265)
        at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:727)
        at io.quarkus.smallrye.openapi.deployment.SmallRyeOpenApiProcessor.getUserDefinedFilters(SmallRyeOpenApiProcessor.java:451)
        at io.quarkus.smallrye.openapi.deployment.SmallRyeOpenApiProcessor.getUserDefinedRuntimeFilters(SmallRyeOpenApiProcessor.java:432)
        at io.quarkus.smallrye.openapi.deployment.SmallRyeOpenApiProcessor.registerAnnotatedUserDefinedRuntimeFilters(SmallRyeOpenApiProcessor.java:237)
        at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:735)
        at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:856)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:256)
        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2675)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2654)
        at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1627)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1594)
        at java.base/java.lang.Thread.run(Thread.java:1575)
        at org.jboss.threads.JBossThread.run(JBossThread.java:499)

        at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:355)
        at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:272)
        at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:62)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:89)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:428)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:55)
        at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:138)
        at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:93)
        at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:131)
        at io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:62)
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.smallrye.openapi.deployment.SmallRyeOpenApiProcessor#registerAnnotatedUserDefinedRuntimeFilters threw an exception: java.lang.NullPointerException: Cannot invoke "org.jboss.jandex.AnnotationValue.asEnum()" because the return value of "org.jboss.jandex.AnnotationInstance.value()" is null
        at io.quarkus.smallrye.openapi.deployment.SmallRyeOpenApiProcessor.lambda$getUserDefinedFilters$4(SmallRyeOpenApiProcessor.java:446)
        at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:196)
        at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1709)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:570)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:560)
        at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:265)
        at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:727)
        at io.quarkus.smallrye.openapi.deployment.SmallRyeOpenApiProcessor.getUserDefinedFilters(SmallRyeOpenApiProcessor.java:451)
        at io.quarkus.smallrye.openapi.deployment.SmallRyeOpenApiProcessor.getUserDefinedRuntimeFilters(SmallRyeOpenApiProcessor.java:432)
        at io.quarkus.smallrye.openapi.deployment.SmallRyeOpenApiProcessor.registerAnnotatedUserDefinedRuntimeFilters(SmallRyeOpenApiProcessor.java:237)
        at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:735)
        at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:856)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:256)
        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2675)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2654)
        at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1627)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1594)
        at java.base/java.lang.Thread.run(Thread.java:1575)
        at org.jboss.threads.JBossThread.run(JBossThread.java:499)

        at io.quarkus.builder.Execution.run(Execution.java:124)
        at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:79)
        at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:161)
        at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:351)
        ... 9 more
Caused by: java.lang.NullPointerException: Cannot invoke "org.jboss.jandex.AnnotationValue.asEnum()" because the return value of "org.jboss.jandex.AnnotationInstance.value()" is null
        at io.quarkus.smallrye.openapi.deployment.SmallRyeOpenApiProcessor.lambda$getUserDefinedFilters$4(SmallRyeOpenApiProcessor.java:446)
        at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:196)
        at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1709)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:570)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:560)
        at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:265)
        at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:727)
        at io.quarkus.smallrye.openapi.deployment.SmallRyeOpenApiProcessor.getUserDefinedFilters(SmallRyeOpenApiProcessor.java:451)
        at io.quarkus.smallrye.openapi.deployment.SmallRyeOpenApiProcessor.getUserDefinedRuntimeFilters(SmallRyeOpenApiProcessor.java:432)
        at io.quarkus.smallrye.openapi.deployment.SmallRyeOpenApiProcessor.registerAnnotatedUserDefinedRuntimeFilters(SmallRyeOpenApiProcessor.java:237)
        at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:735)
        at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:856)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:256)
        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2675)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2654)
        at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1627)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1594)
        at java.base/java.lang.Thread.run(Thread.java:1575)
        at org.jboss.threads.JBossThread.run(JBossThread.java:499)


### How to Reproduce?


[oas-filter-no-ann.zip](https://github.com/user-attachments/files/18585310/oas-filter-no-ann.zip)

unzip
mvn quarkus:dev

### Output of `uname -a` or `ver`

MINGW64_NT-10.0-26100 NANBCHL9NG3 3.3.6-341.x86_64 2022-09-05 20:28 UTC x86_64 Msys

### Output of `java -version`

openjdk 23.0.2 2025-01-21 OpenJDK Runtime Environment Temurin-23.0.2+7 (build 23.0.2+7) OpenJDK 64-Bit Server VM Temurin-23.0.2+7 (build 23.0.2+7, mixed mode, sharing)

### Quarkus version or git rev

3.17.2, 3.17.8, 3.18.0

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.9 (8e8579a9e76f7d015ee5ec7bfcdc97d260186937) Maven home: C:\eclipse\tools\java\maven Java version: 23.0.2, vendor: Eclipse Adoptium, runtime: C:\eclipse\tools\java\23 Default locale: de_DE, platform encoding: UTF-8 OS name: "windows 11", version: "10.0", arch: "amd64", family: "windows"

### Additional information

_No response_

--- a/extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java
+++ b/extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java
@@ -443,7 +443,7 @@ public class SmallRyeOpenApiProcessor {
         return index
                 .getAnnotations(OpenApiFilter.class)
                 .stream()
-                .filter(ai -> stages.contains(OpenApiFilter.RunStage.valueOf(ai.value().asEnum())))
+                .filter(ai -> stages.contains(OpenApiFilter.RunStage.valueOf(ai.valueWithDefault(index).asEnum())))
                 .sorted(comparator)
                 .map(ai -> ai.target().asClass())
                 .filter(c -> c.interfaceNames().contains(DotName.createSimple(OASFilter.class.getName())))

